### ISSUE TITLE
[GTK] Dialog close closes invisible main window

### ISSUE THREAD

**Issue by MartinRothschink** [picoe/Eto#1132]
Still working on my tray application with an invisible main  window. I added a separate about dialog which is created from the tray context menu. Closing this dialog closes the entire application on GTK. MAC and WPF are fine.

## Expected Behavior
Closing my about dialog should not close the application if main window is invisible.

## Actual Behavior
Closing my about dialog dialog closes invisible main window and application.

## Steps to reproduce the issue

1. Run code below, only tray icon is visible.
2. Click on tray icon to open context menu, select About...
3. About dialog is displayed, close dialog 
4. Application exists

## Code that Demonstrates the Problem

```
using System;
using Eto.Forms;
using Eto.Drawing;

namespace EtoTrayApp
{
   public class About : Dialog
   {
      public About()
      {
         Title = "My About Dialog";
         ClientSize = new Size(400, 350);

         Content = new StackLayout
         {
            Padding = 10,
            Items =
            {
               "About!",
            }
         };
      }
   }

   public class MainForm : Form
   {
      private bool _startup = true;
      private TrayIndicator _tray;
      public MainForm()
      {
         Title = "My Eto Form";
         ClientSize = new Size(200, 200);
         _tray = new TrayIndicator
         {
            Image = Eto.Drawing.Icon.FromResource("EtoTrayApp.error.ico"),
            Menu = new ContextMenu
            {
               Items =
               {
                  new ButtonMenuItem {Text = "About...", Command = new Command((s, e) =>
                  {
                     var about = new About();
                     about.ShowModal(this);
                  })},
                  new ButtonMenuItem {Text = "Quit", Command = new Command((s, e) => Application.Instance.Quit())}
               }
            }
         };

         ShowInTaskbar = false;
         _tray.Visible = true;
      }

      protected override void OnShown(EventArgs e)
      {
         base.OnShown(e);
         if (_startup)
         {
            // change this to false to show the issue
            Visible = false;
            _startup = false;
         }
      }
   }
}
```

## Specifications

  - Version: Eto.Gtk 2.4.1
  - Platform(s): Linux Gtk
  - Operating System(s): Ubuntu 18.04 with Mono 5.12

**Comment by MartinRothschink** [comment]
Sample project https://1drv.ms/u/s!Ajb182QihKkLs8h16lv-1MzWq0H30A

**Comment by cra0zy** [comment]
You should not be allowing the window to get closed, this is normal...

Here is an example on how what you are trying to achieve should be done: https://gist.github.com/cra0zy/ee7a89a0495f55465935f85859f121f0

**Comment by MartinRothschink** [comment]
Thanks for the sample. I have extended your sample with an about dialog similar to my code above.
https://gist.github.com/MartinRothschink/c31ad3b6fc83b2fed4a128310c9beb3a

Unfortunately it shows the same issue on GTK: If you display the about dialog while the main window is hidden, then close the about dialog, the application exits. This happens regardless if ShowModal() is called with owner of _form or empty.

Moreover there are additional issues on the other platforms. On WPF you get an exception if you try to show the  about dialog without first making the main window visible (seems it's not completely created). Playing with the sample on my MAC I you quickly get a situation where the tray icon does no longer accept mouse clicks (application seems to hang).

The only workaround I found for GTK is to make the main window visible before I show the about dialog.
